addons/l10n_in_ewaybill_stock/tests/test_ewaybill_stock.py

```diff
@@ -102,3 +106,62 @@ def test_ewaybill_stock(self):
             'totInvValue': 2625.0
         }
         self.assertDictEqual(ewaybill._ewaybill_generate_direct_json(), expected_json)
+
+    @freeze_time('2024-04-26')
+    def test_ewaybill_stock_sub_type_other(self):
+        delivery_picking = self._create_stock_picking()
+        ewaybill_form = Form(self.env['l10n.in.ewaybill'].with_context(default_picking_id=delivery_picking.id))
```

Comment:
I would like to go with the form at least it will validates the front end too.

Reply:
As it is important that when the sub type is `others` the description always goes with it.

Reply:
@william-andre Still ok to use Form for this case as the thing sometimes behaved weirdly?

Reply:
For should be used to test UI features, but without the UI.
For instance: onchange, warnings, and the edition of new records.
Not sure what is tested in this case, but since the form emulation is slower than the raw create/write it is usually better to just call these otherwise.